cosmwasm/ibc-union/lightclient/ethereum/src/client.rs

```diff
@@ -323,19 +345,51 @@ fn update_state<C: ChainSpec>(
         }
     }
 
-    Ok((
-        updated_height,
-        ClientState::V1(client_state),
-        consensus_state,
-    ))
+    let mut state_update = StateUpdate::new(updated_height, consensus_state);
+    if client_state.latest_height == consensus_update.finalized_header.execution.block_number {
+        state_update = state_update.set_client_state(ClientState::V1(client_state));
+    }
+
+    if let LightClientUpdate::EpochChange(update) = &mut header.consensus_update {
+        let current_epoch =
+            compute_epoch_at_slot::<C>(update.update_data.finalized_header.beacon.slot);
+        inverse_sync_committee_pubkeys(&update.next_sync_committee);
+        state_update = state_update.add_storage_write(
+            sync_committee_store_key(current_epoch + 1),
+            &update.next_sync_committee,
+        );
+    }
+
+    Ok(state_update)
+}
+
+/// Additive inverse the public keys to do inverse aggregation
+fn inverse_sync_committee_pubkeys(sync_committee: &SyncCommittee) -> SyncCommittee {
```

Comment:
we may want to consider a new type for this to avoid confusion, `InverseSyncCommittee`

Reply:
I agree